### PR TITLE
TG Sprite Project: R&D, Mechfabs, Telecomms, and Station Objects

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Updates the Mechfabs, telecomms, R&D, and station objects items.

Station objects includeds a lot of weird odds and ends things.

Also cleans up the structure of some files that were located where they should be.

img
img
img
img
img
img
img
img
img
img
img
img

:cl: Fox McCloud
imageadd: updates a ton of various station object sprites, R&D, servers, telecomms, and a few other odds and ends
/:cl:

### CODE REVIEW

File: code/game/machinery/tcomms/_base.dm [hunk 1]
```diff
@@ -30,7 +30,7 @@ GLOBAL_LIST_EMPTY(tcomms_machines)
 /obj/machinery/tcomms
 	name = "Telecommunications Device"
 	desc = "Someone forgot to say what this thingy does. Please yell at a coder"
-	icon = 'icons/obj/tcomms.dmi'
+	icon = 'icons/obj/machines/telecomms.dmi'
 	icon_state = "error"
```
Comment (AffectedArc07) [inline code/game/machinery/tcomms/_base.dm:34]: This wont have the icon, since it wasnt carried over

File: code/game/machinery/tcomms/core.dm [hunk 2]
```diff
@@ -13,7 +13,7 @@
 /obj/machinery/tcomms/core
 	name = "Telecommunications Core"
 	desc = "A large rack full of communications equipment. Looks important."
-	icon_state = "core"
+	icon_state = "hub"
```
Comment (AffectedArc07) [inline code/game/machinery/tcomms/core.dm:16]: I want to see if I can get the core sprite I did up to TG standards before nuking it entirely

**Comment by SabreML** [comment]
The RnD/Robotics/Autolathe sprites from TG are the ones I dislike the most by a long shot. I don't know what it is about them but they genuinely make me feel uncomfortable.
Everything else in this though *(Excluding maybe the portals)* looks amazing. So 7/10 from me.

**Comment by TDSSS** [comment]
Not a fan of those portals, the rest look pretty good. But yeah, do those portals look very weird to anyone else?

**Comment by Qwertytoforty** [comment]
> 
> 
> Not a fan of those portals, the rest look pretty good. But yeah, do those portals look very weird to anyone else?

They do seem off in a way, yeah

**Comment by Cocacolagua** [comment]
they're brighter than our current ones

**Comment by Spacemanspark** [comment]
Going to agree with sabre and tdss, not a fan of the rnd and fab stuff, and the portals look really wonky.

**Comment by MRgaty** [comment]
I think our portals look even better, those look super cartoony

**Comment by Dovixx** [comment]
Gonna have to agree with what was said above. The RnD and the mech fabricator machines look weird and I am not a huge fan of those and something about the portals seems weird to me as well.

**Comment by Varghh** [comment]
Portals look terrible, R&D machines too. 
Doesn't matter how many of us voice how bad they look, this will get merged anyway.

**Comment by SabreML** [comment]
The issue with the portals I think is almost definitely the dark outlines, but they do look a lot better in motion than in a still image.
***Current/New:***
Blue Blue Portal
Orange Orange Portal
Still maybe a bit odd, but better looking than the current ones.

**Comment by Mitchs98** [comment]
Agreeing with the above as well. RnD stuff looks a bit whack to me, portals too. Everything else seems to be an improvement.

**Comment by TDSSS** [comment]
Okay, portals look much better in motion.

**Comment by Spacemanspark** [comment]
The outline on the proposed portal sprites _is_ really weird. If it weren't for that I'd probably be fine with the change.

File: code/modules/research/rdmachines.dm [hunk 3]
```diff
@@ -122,15 +122,14 @@
...
 		use_power(MINERAL_MATERIAL_AMOUNT / 10)
 	else
 		var/obj/item/stack/S = type_inserted
 		stack_name = initial(S.name)
 		use_power(min(1000, (amount_inserted / 100)))
-	overlays += "[initial(name)]_[stack_name]"
-	sleep(10)
-	overlays -= "[initial(name)]_[stack_name]"
+	add_overlay("protolathe_[stack_name]")
+	addtimer(CALLBACK(src, /atom/proc/cut_overlay, "protolathe_[stack_name]"), 10)
```
Comment (dearmochi) [inline code/modules/research/rdmachines.dm:132]:
```suggestion
	addtimer(CALLBACK(src, /atom/proc/cut_overlay, "protolathe_[stack_name]"), 1 SECONDS)
```

**Review by dearmochi — APPROVED** [review]
The following icon states have rogue pixels:
- `broadcast receiver`
- `broadcast receiver_o`
- `broadcaster_send`
- `hub`
- `hub_o`
- `comm_server`
- `comm_server_o`
- `tdoppler`
- `circuit_imprinter_ani`
- `protolathe_n`
- `RD-server-halt`
- `autolathe_n` (it also seems to lose a chunk near the end)
- `mechfab3`

Not super bad, so these could be ignored and fixed in a later PR (and backported if necessary)

**Comment by AffectedArc07** [comment]
> The following icon states have rogue pixels:
> 
> * `broadcast receiver`
> * `broadcast receiver_o`
> * `broadcaster_send`
> * `hub`
> * `hub_o`
> * `comm_server`
> * `comm_server_o`
> * `tdoppler`
> * `circuit_imprinter_ani`
> * `protolathe_n`
> * `RD-server-halt`
> * `autolathe_n` (it also seems to lose a chunk near the end)
> * `mechfab3`
> 
> Not super bad, so these could be ignored and fixed in a later PR (and backported if necessary)

Always verify rogue pixels locally, IDB has this as a known issue when it doesnt exist on the sprites
image

**Comment by Denthamos** [comment]
The machinery is not that bad, but the portals just look ....off.  Too bright and pastel white for both colors.

**Comment by McRamon** [comment]
Problem with those machines people are against - if i am not mistaken, TG took them from Eris which has very unique and distinguish art direction. Either way there is no reason to discuss anything about new sprites in my opinion, first we get consistent perspective on all objects in game, then we can talk about improving what community think is not great (and there are a lot of things on TG that are not great).

**Review by Kyep — APPROVED** [review]
While I am not keen on blindly copying anything (including sprites) from TG (or any other server):

- I do feel like most of these sprites are obvious upgrades.
- I also feel like it doesn't make sense for us to use different sprites for the exact same machine without a good reason for doing so.

Based on the above two reasons, I'm approving this.

File: code/modules/research/rdmachines.dm (hunk 3 above)
Comment (AffectedArc07) [reply code/modules/research/rdmachines.dm:132]: Once this is addressed ill merge

**Review by AffectedArc07 — APPROVED** [review]
Consistency is consistency, and we can always have some sprites resprite things in this style if these sprites are *that* bad.

**Pull request merged**